actions/QuizResultsAction.php

```diff
@@ -0,0 +1,189 @@
+            $results = $this->quizManager->keepOnlyBestResult($results);
+        }
+        if (!$this->arguments['rawdata']) {
+            // put objects in results
+            $coursesCached = [];
+            $modulesCached = [];
+            $activitiesCached = [];
+            $learnersCached = [];
+            $results = array_map(function ($result) use ($coursesCached, $modulesCached, $activitiesCached, $learnersCached) {
+                if (!isset($coursesCached[$result['course']])) {
+                    $coursesCached[$result['course']] = $this->courseManager->getCourse($result['course']);
+                }
+                $result['course'] = $coursesCached[$result['course']];
+                if (!isset($modulesCached[$result['module']])) {
+                    $modulesCached[$result['module']] = $this->courseManager->getModule($result['module']);
+                }
+                $result['module'] = $modulesCached[$result['module']];
+                if (!isset($activitiesCached[$result['activity']])) {
+                    $activitiesCached[$result['activity']] = $this->courseManager->getActivity($result['activity']);
+                }
+                $result['activity'] = $activitiesCached[$result['activity']];
+                if (!isset($learnersCached[$result['learner']])) {
+                    $learnersCached[$result['learner']] = $this->learnerManager->getLearner($result['learner']);
+                }
+                $result['learner'] = $learnersCached[$result['learner']];
+                $result['log_time'] = $this->dateManager->createDatetimeFromString($result['log_time']);
+                return $result;
+            }, $results);
+
+            if ($this->arguments['noadmins'] && !$this->params->get('lms_config')['save_progress_for_admins']) {
+                $results = array_filter($results, function ($result) {
+                    return !($result['learner']->isAdmin());
+                });
+            }
+        } elseif ($this->arguments['noadmins'] && !$this->params->get('lms_config')['save_progress_for_admins']) {
+            $results = array_filter($results, function ($result) {
+                return !($this->learnerManager->getLearner($result['learner'])->isAdmin());
+            });
+        }
+
+        $urlParams = [];
+        foreach (['course','module','activity','quizId','learner','rawdata','onlybest','noadmins','content'] as $param) {
+            if (!empty($this->arguments[$param])) {
+                $urlParams[$param] = $this->arguments[$param];
+            }
+        }
+
+        return $this->render(
+            '@lms/quiz-results.twig',
+            [
+                'results' => $results,
+                'rawdata' => $this->arguments['rawdata'] ,
+                'urlParams' => $urlParams,
+                'handler' => !empty($this->arguments['content']) ? 'render' : null,
+                'dateFormat'=>DateManager::DATETIME_FORMAT,
```

Comment:
détail que j'avais noté, tu n'as pas besoin de passer le format, tu y accéder directement à la constante de classe depuis Twig, exemple : {{ constant('Acme\\DemoBundle\\Entity\\Demo::MY_CONSTANT') }}

Reply:
trop cool merci

Reply:
Corrigé par https://github.com/YesWiki/yeswiki-extension-lms/commit/047c71fab512e5307c5ea1143eda206f9273f4a2